pkgs/racket-test-core/tests/racket/hash.rktl

```diff
@@ -123,6 +123,70 @@
          #hash([one . 1] [two . 2] [three . 3] [four . 4]))
         h))
 
+;; ----------------------------------------
+;; hash-filter-key and hash-filter-value:
+
+;; filtering by key with strings
+(test #hash(["one" . 1])
+      hash-filter-keys
+      #hash(["one" . 1] ["two" . 2] ["three" . 3])
+      (λ (k) (string=? k "one")))
+
+;; filtering by value with strings
+(test #hash(['carrot . "vegetable"])
```

Comment:
I think this test is wrong. Shouldn't it have `['apple . "fruit"]` and `['banana . "fruit"]` in the resulting hash?

Also, the key here is `'(quote carrot)`, not `'carrot`. I would recommend using `(hash key1 val1 key2 val2)` instead of `#hash(...)`.

Reply:
Sorry for misleading! I somehow thought it's filtering _out_.